R/ggMarginalHelpers.R

```diff
@@ -90,121 +151,87 @@ addMainTheme <- function(marginal, margin, p) {
       marginUnit <- "null"
       marginLength <- 0
     } else {
-      marginUnit <- attr(p$theme$axis.ticks.length, "unit")
-      marginLength <- as.numeric(p$theme$axis.ticks.length, "unit")
+      marginUnit <- attr(scatPTheme$axis.ticks.length, "unit")
+      marginLength <- as.numeric(scatPTheme$axis.ticks.length, "unit")
     }
-    if (margin == "x") {
-      marginal <- marginal + 
+    
+    if (marg == "x") {
+      rawMarg <- rawMarg + 
         ggplot2::theme(
           axis.title.x = ggplot2::element_blank(),
           axis.text.x = ggplot2::element_blank(),
           plot.margin = grid::unit(c(0, 0, 0, marginLength), marginUnit)
         )
     } else {
-      marginal <- marginal + 
+      rawMarg <- rawMarg + 
         ggplot2::theme(
           axis.title.y = ggplot2::element_blank(),
           axis.text.y = ggplot2::element_blank(),
           plot.margin = grid::unit(c(0, 0, marginLength, 0), marginUnit)
         )
     }
-  }
-  # if this is the old ggplot2 version, things are simpler
-  else {
-    marginal <- marginal +
-      ggplot2::theme(
-        text = ggplot2::element_text(size = p$theme$text$size, color = "transparent"),
-        line = ggplot2::element_blank(),
-        panel.background = ggplot2::element_blank(),
-        axis.text = ggplot2::element_text(color = "transparent")
-      )
-    
-    if (margin == "x") {
-      marginal <- marginal + 
-        ggplot2::theme(
-          axis.title.x = ggplot2::element_blank(),
-          axis.text.x = ggplot2::element_blank(),
-          plot.margin = grid::unit(c(0, 0, -1, 0), "lines")
-        )
-    } else {
-      marginal <- marginal + 
-        ggplot2::theme(
-          axis.title.y = ggplot2::element_blank(),
-          axis.text.y = ggplot2::element_blank(),
-          plot.margin = grid::unit(c(0, 0, 0, -1), "lines")
-        )
-    }
-  }
   
-  marginal
+  rawMarg
 }
 
 # Copy the scale transformation from the original plot (reverse/log/limits/etc)
 # We have to do a bit of a trick on the marginal plot that's flipped by
 # taking the original x/y scale and manually changing it to the other axis
-getScale <- function(margin, type, pb) {
-  if (margin == "x") {
-    if (type == "boxplot") {
-      scale <- pb$layout$panel_scales$x[[1]]
-      scale$aesthetics <- gsub("^x", "y", scale$aesthetics)
-      scale$limits <- pb$layout$panel_scales$x[[1]]$get_limits()
-    } else {
-      scale <- pb$layout$panel_scales$x[[1]]
-    }
-  } else if (margin == "y") { 
-    if (type == "boxplot") {
-      scale <- pb$layout$panel_scales$y[[1]]
-      scale$limits <- pb$layout$panel_scales$y[[1]]$get_limits()
-    } else {
-      scale <- pb$layout$panel_scales$y[[1]]
-      scale$aesthetics <- gsub("^y", "x", scale$aesthetics)
-    }
+getScale <- function(marg, type, builtP) {
+  
+  scale <- getPanelScale(marg = marg, builtP = builtP)
+  
+  if (type == "boxplot") {
+    scale$limits <- scale$get_limits()
+  }
+  
+  if (needsFlip(marg = marg, type = type)) {
```

Comment:
I think if we actually cared about the data in y then it would matter, but since we don't, it doesn't matter. We only even need y when its a boxplot, and even then, as long as it's non-null then it doesn't matter what the values are (I think?).